dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/generator/GenerateTagFileVisitor.java

```diff
@@ -318,7 +318,13 @@ protected void generateInitSection(ValidateTagFileResult validatorResult) throws
             writer.print(");");
             writer.println();
         }
-        if (PagesVersionHandler.isPages31OrHigherLoaded()) {
+        if (PagesVersionHandler.isPages40OrHigherLoaded()) {
+            if (aliasSeen) {
+                writer.println("this.jspContext = new org.apache.jasper.runtime.JspContextWrapper(this, ctx, _jspx_nested, _jspx_at_begin, _jspx_at_end, aliasMap);");
+            } else {
+                writer.println("this.jspContext = new org.apache.jasper.runtime.JspContextWrapper(this, ctx, _jspx_nested, _jspx_at_begin, _jspx_at_end, null);");
+            }
+        } else if (PagesVersionHandler.isPages31Loaded()) {
```

Comment:
? Let's follow up here so we know why we made this change, perhaps add a comment?

Reply:
constructor changed for new code pulled in